flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -199,6 +214,85 @@ protected static boolean allVerticesWithinUtilizationTarget(
         return true;
     }
 
+    protected static boolean resourceQuotaReached(
+            Configuration conf,
+            EvaluatedMetrics evaluatedMetrics,
+            Map<JobVertexID, ScalingSummary> scalingSummaries,
+            JobAutoScalerContext<?> ctx) {
+
+        if (evaluatedMetrics.getJobTopology() == null
+                || evaluatedMetrics.getJobTopology().getSlotSharingGroupMapping().isEmpty()) {
+            return false;
+        }
+
+        var cpuQuota = conf.getOptional(AutoScalerOptions.CPU_QUOTA);
+        var memoryQuota = conf.getOptional(AutoScalerOptions.MEMORY_QUOTA);
+        var tmMemory = ctx.getTaskManagerMemory();
+        var tmCpu = ctx.getTaskManagerCpu();
+
+        if (cpuQuota.isPresent() || memoryQuota.isPresent()) {
+            var currentSlotSharingGroupMaxParallelisms = new HashMap<SlotSharingGroupId, Integer>();
+            var newSlotSharingGroupMaxParallelisms = new HashMap<SlotSharingGroupId, Integer>();
+            for (var e :
+                    evaluatedMetrics.getJobTopology().getSlotSharingGroupMapping().entrySet()) {
```

Comment:
We should have some workaround similarly to the `scalingWouldExceedClusterResources` method which checks whether the maximum found parallelism is identical to the total number of task slots to determine that slot sharing is enabled and otherwise assume it isn't.

Reply:
Such case Job topology is null so it's handled: https://github.com/apache/flink-kubernetes-operator/pull/789/files#diff-7fdb929157b6a94cc180c67b8dddb6722d26a7b8ea8259eb7cb9ef84b9a418a3R224-R227